### DISCUSSION TITLE
Analyse Error Details

### DISCUSSION THREAD

**Discussion by MoritzSalein** [ludeknovy/jtl-reporter#90] [Ideas]
In some situation it'll be helpful to see the error details of the failed requests. The details should give the total count of the errors with the response code and response message if it was stored with the jtl report file (in my case with JMeter). 
In my tests I haven't stored an extra error report file, which I can get additional information about error requests.

**Comment by ludeknovy** [comment]
Hi @MoritzSalein again. This feature is somehow present, but I will need to refactor it completely (you need to upload special file with errors, but its no very well handled). So far I removed it for the next major version. 

In the next iteration of this feature, I will take the data from `response_message` and make it thus available even for Locust.

**Comment by MoritzSalein** [comment]
@ludeknovy I worked with JMeter report files, where I included the response data on error (that a configuration in jmeter.properties called jmeter.save.saveservice.response_data.on_error. Perhaps something for later documentation.) But I'm not sure, if I'll get different information, if I use a csv output instead a xml. 
In my current generated jtl files I get the values of `responseCode`, `responseMessage` and `failureMessage` columns and in best you can show this information in a report.

**Reply by ludeknovy** [reply]
Status codes

https://user-images.githubusercontent.com/13610612/140476940-2e771fef-559b-4490-bcbe-480078da3268.mov

**Reply by ludeknovy** [reply]
Status codes and response messages

https://user-images.githubusercontent.com/13610612/140488344-c4250382-ea9a-4ead-a207-86710bf205a3.mov

**Reply by MoritzSalein** [reply]
Sorry for my late response. I looked at your solutions and I have following replies:
to Video 1
There aren't enough information for my analyses. 

to Video 2
That the best idea. But in the graph of the Response Codes there must be always the numeric codes of the HTTP status. Only in the table it will be helpful to see the message as you recorded.

**Reply by ludeknovy** [reply]
As you already know, the 2) was implemented. The non-numeric status code is actually provided by JMeter 🤷 .

**Reply by MoritzSalein** [reply]
Yes, but what I mean is, that I want to have a table with all different error codes, the error message (including the assertion) and the count of error-requests.
For the assertion notice I add an issue https://github.com/ludeknovy/jtl-reporter/issues/114

And in the top of the page, it will be nice to have a bar graph, with the different status codes and its visualisation, how often you got the different codes.

**Reply by ludeknovy** [reply]
So the table then should contain: response failure, assertion error, status code, occurrence (count) ?

About the graph - I think the tree chart does a good job in this case. Why do you think the bar chart would be better?

**Reply by MoritzSalein** [reply]
to 1: yes, that are the values
to 2: sorry for my misunderstanding word bar chart. The current graph is the wright one.
